src/DynamoCore/Scheduler/CompileTask.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using Dynamo.Engine.CodeGeneration;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Interfaces;
+
+namespace Dynamo.Scheduler
+{
+    /// <summary>
+    /// A task for compiling the graph to Expressions.
+    /// </summary>
+    public class ExpressionCompileTask : AsyncTask
+    {
+        private readonly IEnumerable<NodeModel> nodes;
+        private readonly NodeCompilerRegistry<Expression> compilers;
+        private Dictionary<Guid, List<Expression>> nodeExpressionMap = new Dictionary<Guid, List<Expression>>();
+         
+        public ExpressionCompileTask(IScheduler scheduler, WorkspaceModel workspace, NodeCompilerRegistry<Expression> compilers) : base(scheduler)
+        {
+            this.nodes = workspace.Nodes;
+            this.compilers = compilers;
+        }
+
+        public override TaskPriority Priority
+        {
+            get { return TaskPriority.Normal; }
+        }
+
+        protected override void HandleTaskExecutionCore()
+        {
+            // For each node, look for the upstream nodes, and
+            // get their output port identifiers.
+            var sortedNodes = AstBuilder.TopologicalSort(nodes).ToArray();
+
+            nodeExpressionMap.Clear();
+            foreach (var node in sortedNodes)
+            {
+                nodeExpressionMap.Add(node.GUID, BuildNodeExpressions(node));
+            }
+
+            var block = Expression.Block(nodeExpressionMap.Values.Last());
```

Comment:
This just gets the last of the expression from the topologically sorted nodes. Again, this is dumb. This would fail to compile multiple branches with output nodes.